client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -109,7 +109,7 @@ public static Command from(
         } else {
             final ResourceIdentifier addressIdentifier = ResourceIdentifier.fromString(message.getAddress());
             if (!tenantId.equals(addressIdentifier.getTenantId())) {
-                validationErrorJoiner.add("address contains wrong tenant");
+                validationErrorJoiner.add("address contains wrong tenant '" + addressIdentifier.getTenantId() + "'");
```

Comment:
well, wouldn't it make sense to include both the expected tenant as well as the found (wrong) one?

Reply:
In my case the `Command.toString()` was used, which already contained the expected tenant. That was mixing me up. Because you get a error message, telling the wrong tenant is used, but displays the expected tenant. 
So, I'm not sure, if adding the expected tenant makes sense for other cases. For my case, it's not required.

Reply:
I think adding the wrong tenant is enough. `Command.toString()` output would then be something like:
`Invalid Command [tenant-id: myTestTenant, device-id: myTestDevice. error: address contains wrong tenant: 'tenantFromAddress']`